File: 01-ordenacao/csharp/OrdenacaoMVC/src/Model/ResultadoOrdenacao.cs
namespace OrdenacaoMVC.Model;

/// <summary>
/// Métricas de uma execução de ordenação
///</summary>
/// <param name="Comparacoes">Quantidade de comparações entre elementos</param>
/// <param name="Trocas">Quantidade de trocas (swap de dois elementos)</param>
/// <param name="Movimentacoes">Quantidade de movimentações/deslocamentos (Insertion Sort shifts)</param>
/// <param name="TempoMs">Tempo de execução em milissegundos</param>
/// <param name="TamanhoLista">Tamanho da lista ordenada</param>
public record ResultadoOrdenacao(
    long Comparacoes,
    long Trocas,
    long Movimentacoes,
    long TempoMs,
    int TamanhoLista);

File: 01-ordenacao/java/OrdenacaoMVC/src/model/ResultadoOrdenacao.java
package model;

/**
 * Métricas de uma execução de ordenação
 *
 * @param comparacoes quantidade de comparações
 * @param trocas quantidade de trocas (swap de dois elementos)
 * @param movimentacoes quantidade de movimentações/deslocamentos (Insertion Sort shifts)
 * @param tempoMs tempo em milissegundos
 * @param tamanhoLista tamanho da lista ordenada
 */
public record ResultadoOrdenacao(// record é imutável, não precisa de getters/setters
        long comparacoes,
        long trocas,
        long movimentacoes,
        long tempoMs,
        int tamanhoLista) {
}
